src/main/java/edu/harvard/iq/dataverse/engine/command/impl/ReconcileDatasetPidCommand.java

```diff
@@ -0,0 +1,181 @@
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Given a draft datasetVersion (which has no published predecessors), and PID provider that is different to the current
+ * configured one this command reconfigures the PID provider, and updates the PID to the new provider (Removed the
+ * existing PID, registers a new one and notifies the user about the change).
+ *
+ * @author jdarms
+ */
+// No required permissions because we check for superuser status.
+@RequiredPermissions({})
+public class ReconcileDatasetPidCommand extends AbstractDatasetCommand<Dataset> {
+
+    private static final Logger logger = Logger.getLogger(ReconcileDatasetPidCommand.class.getName());
+    PidProvider newPidProvider;
+
+    public ReconcileDatasetPidCommand(DataverseRequest aRequest, Dataset theDataset, PidProvider newPidProvider) {
+        super(aRequest, theDataset);
+        this.newPidProvider = newPidProvider;
+    }
+
+
+    @Override
+    public Dataset execute(CommandContext ctxt) throws CommandException {
+        // ensure that only superuser can execute the command.
+        if (!(getUser() instanceof AuthenticatedUser) || !getUser().isSuperuser()) {
+            throw new PermissionException(BundleUtil.getStringFromBundle("admin.api.auth.mustBeSuperUser"), this,
+                    Collections.singleton(Permission.EditDataset), getDataset());
+        }
+        // Datast must be unreleased! This means there is only one version!
+        if (getDataset().isReleased()) { //@TODO: Clarify whether this is the best check...
+            throw new IllegalCommandException("Dataset already published, cannot alter PID Provider", this);
+        }
+        // Dataset must not be harvested!
+        if (getDataset().isHarvested()) {
+            throw new IllegalCommandException("Dataset is harvested, cannot alter PID Provider", this);
+        }
+        PidProvider currentPidProvider = getDataset().getEffectivePidGenerator();
+        // new PID Provider must be different to requested one!
+        if (this.newPidProvider.equals(currentPidProvider)) {
+            throw new IllegalCommandException("PID Provider " + currentPidProvider.getId() + " is same as configured. This Operation has no effect!", this);
+        }
+
+        GlobalId oldId = getDataset().getGlobalId();
+        if (oldId == null) {
+            throw new IllegalStateException("Dataset without a global identifier, cannot alter!");
+        }
+        logger.fine("Reconciling dataset( id =`" + getDataset().getId() + ")` - removing globalId `" + getDataset().getGlobalId() + '`');
+        // remove dataset PID
+        try {
+            if (currentPidProvider.alreadyRegistered(getDataset())) { //if not registered with PIDProvider than there is no need to delete it...
+                currentPidProvider.deleteIdentifier(getDataset()); // delete it externally
```

Comment:
Perhaps it would be better to not delete this, so it remains reserved and can't be used again for new datasets? (Doesn't help for PermaLinks, but there (or for all providers) you could extend the local PID lookup to also check the alternate id field.)

Reply:
While this would resolve some of our problems we end up with a bunch of registered PIDs that will never be used. I'd rather accept the risk (I'm not sure if many are using this feature) and document that an administrator must complete the reconciliation offline, i.e. move the files and delete the alternativeIdentifier from the database. 

> you could extend the local PID lookup to also check the alternate id field.
Sure do, I believe this is needed anyhow.

Reply:
Still needs to be documented

Reply:
How about adding a second call to a new AlternativePersistentIdentifier.findByProtocolIdentifierAuthority query in the DVObjectServiceBean.isGlobalIdLocallyUnique method (only if the first query would return true)? (Query could be modelled on DvObject.findIdByAlternativeGlobalId query but avoiding the join since we don't care which dvObject has that alt PID). That would assure that the old id is never reused. (Seems like a generally good idea even though your code may be the first automated way to end up with an alt PID from an active provider. It's possible that someone migrating in data from elsewhere could be reusing

Reply:
Done as suggested, alternatives identifiers are not included in the `isGlobalIdLocallyUnique` check of the PIDprovider.